tests/test_1_register_user.py

```diff
@@ -1,159 +1,47 @@
-    register_button = driver.find_element(
-        By.XPATH, '//button[@data-qa="create-account"]'
-    )
-    register_button.click()
-
-    # Verify that registration form is complete
-    account_created_text_present = element_present(
-        driver,
-        By.XPATH,
-        '//h2[@data-qa="account-created"]/b[contains(text(), "Account Created!")]',
-    )
-    assert account_created_text_present == True
-
-    # find continue button and click it
-    continue_button = driver.find_element(By.XPATH, '//a[@data-qa="continue-button"]')
-    continue_button.click()
-
-    # Verify there's label saying "Logged in as ..."
-    logged_in_status = driver.find_element(
-        By.XPATH, '//ul[@class="nav navbar-nav"]/li[last()]/a'
-    )
-    assert (
-        logged_in_status.text
-        == "Logged in as " + user_data["first_name"] + " " + user_data["last_name"]
-    )
-
-    # Find delete account button and click it
-    delete_account_button = driver.find_element(
-        By.XPATH, '//a[@href="/delete_account"]'
-    )
-    delete_account_button.click()
-
-    # verify there's label saying "Account Deleted!"
-    account_deleted_text_present = element_present(
-        driver,
-        By.XPATH,
-        '//h2[@data-qa="account-deleted"]/b[contains(text(), "Account Deleted!")]',
-    )
-    assert account_deleted_text_present == True
-
-    # Press continue button
-    continue_button = driver.find_element(By.XPATH, '//a[@data-qa="continue-button"]')
-    continue_button.click()
+    homepage = HomePage(init_driver)
+    login_page = LoginPage(init_driver)
+    signup_page = SignUpPage(init_driver)
+    account_actions_page = AccountActionPage(init_driver)
+
+    homepage.go_to_website()
+    homepage.check_website_visible()
+    homepage.login()
+
+    login_page.check_new_user_label()
+    login_page.signup_enter_name(user_data["first_name"])
+    login_page.signup_enter_email(user_data["email"])
+    login_page.click_signup()
+
+    signup_page.select_title(user_data["title"])
+    signup_page.enter_name(f"{user_data['first_name']} {user_data['last_name']}")
+    signup_page.enter_password(user_data["password"])
+    signup_page.select_dob(
+        user_data["dob"][0], user_data["dob"][1], user_data["dob"][2]
+    )
+    signup_page.check_newsletter()
+    signup_page.check_special_offers()
+    signup_page.enter_first_name(user_data["first_name"])
+    signup_page.enter_last_name(user_data["last_name"])
+    signup_page.enter_address(user_data["address"])
+    signup_page.select_country(user_data["country"])
+    signup_page.enter_state(user_data["state"])
+    signup_page.enter_city(user_data["city"])
+    signup_page.enter_zipcode(user_data["zipcode"])
+    signup_page.enter_mobile_number(user_data["mobile_number"])
+    signup_page.register()
+
+    account_actions_page.verify_account_created()
+    account_actions_page.click_continue()
+
+    homepage.check_login_status(f"{user_data['first_name']} {user_data['last_name']}")
```

Comment:
here is where your test should end. below you are deleting account - this could be a separate test or a fixture and should not be included inside a testcase

Reply:
I recommend adding some negative tests for registration. Maybe try to enter a mobile number containing letters or name that is too long?